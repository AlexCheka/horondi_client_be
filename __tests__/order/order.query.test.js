/* eslint-disable no-undef */
const { gql } = require('apollo-boost');
const client = require('../../utils/apollo-test-client');
const { newOrder } = require('./order.variables');

let orderId;

describe('Order queries', () => {
  beforeAll(async () => {
    const createOrder = await client.mutate({
      mutation: gql`
        mutation($order: OrderInput!) {
          addOrder(order: $order) {
            ... on Order {
              _id
            }
          }
        }
      `,
      variables: { order: newOrder },
    });
    orderId = createOrder.data.addOrder._id;
  });
  
  test('Should receive all orders', async () => {
    const res = await client.query({
      query: gql`
        query {
          getAllOrders {
            _id
            status
            totalPrice {
              currency
              value
            }
            user {
              address {
                appartment
                city
                region
                street
                country
                buildingNumber
                zipcode
              }
              email
              lastName
              firstName
              phoneNumber
            }
            items {
              bottomColor {
                lang
                value
              }
              closure {
                lang
                value
              }
              model {
                lang
                value
              }
              closureColor 
              size{
                widthInCm
                weightInKg
                heightInCm
                volumeInLiters
                depthInCm
              }
              additions {
                lang
                value
              }
              actualPrice {
                currency
                value
              }
              name {
                lang
                value
              }
              pattern {
                lang
                value
              }
              category {
                lang
                value
              }
              quantity
              colors {
                lang
                value
              }
              subcategory {
                lang
                value
              }
              bottomMaterial {
                lang
                value
              }
            }
            delivery {
              sentBy
              invoiceNumber
            }
            paymentMethod
          }
        }`
    })

    const orders = res.data.getAllOrders;

    expect(orders).toBeDefined();
    expect(orders.length).toBeGreaterThan(0);
    expect(orders).toBeInstanceOf(Array);
    expect(orders.find(order => order._id === orderId)).toMatchObject({
      status:"pending",
      user:{
        __typename: "OrderUser",
        firstName:"Test",
        lastName:"Test",
        email: "test@gmail.com",
        phoneNumber:"380953544271",
        address: {
          __typename: "Address",
          country:"Україна",
          region:"Кіровоградська область",
          city:"Новомиргород",
          zipcode:98908,
          street:"Бульвар Марії Приймаченко",
          buildingNumber:"25",
          appartment:"97"
        }
      },
      delivery:{
        __typename: "Delivery",
        sentBy:"Nova Poshta",
        invoiceNumber:"6280260"
      },
      items:[
        {
          __typename: "OrderItems",
          category:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Сумки"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Bags"
            }
          ],
          subcategory:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Сумки"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Bags"
            }
          ],
          model:
          [
            {
              __typename: "Language",
              lang:"uk",
              value:"Сумка з гобеленом"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Bag with a Pattern"
            }
          ],
          name:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Сумка з гобеленом синя"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Bag with a Pattern Blue"
            }
          ],
          colors:[
            [
              {
                __typename: "Language",
                lang:"uk",
                value:"Сталево-блакитний"
              },
              {
                __typename: "Language",
                lang:"en",
                value:"Steel-blue"
              }
            ]
          ],
          pattern:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Олені"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Deers"
            }
          ],
          closure:[],
          closureColor:"",
          size:{
            __typename: "Size",
            heightInCm:38,
            widthInCm:36,
            depthInCm:10,
            volumeInLiters:0,
            weightInKg:0
          },
          bottomMaterial:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Тканина Кордура"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Cordura fabric"
            }
          ],
          bottomColor:[
            {
              __typename: "Language",
              lang:"uk",
              value:"чорний"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"black"
            }
          ],
          additions:[],
          actualPrice:[
            {
              __typename: "CurrencySet",
              currency:"UAH",
              value:90000
            },
            {
              __typename: "CurrencySet",
              currency:"USD",
              value:3246
            }
          ],
          quantity:1
        }],
        paymentMethod:"card",
        totalPrice:[
          {
            __typename: "CurrencySet",
            currency:"UAH",
            value:90000
          },
          {
            __typename: "CurrencySet",
            currency:"USD",
            value:3246
          }
        ]
      });
  });

  test('should recive order by id', async () => {
    const res = await client.query({
      query: gql`
        query($id: ID!) {
          getOrderById(id: $id) {
            ...on Order {
              status
              totalPrice {
                currency
                value
              }
              user {
                address {
                  appartment
                  city
                  region
                  street
                  country
                  buildingNumber
                  zipcode
                }
                email
                lastName
                firstName
                phoneNumber
              }
              items {
                bottomColor {
                  lang
                  value
                }
                closure {
                  lang
                  value
                }
                model {
                  lang
                  value
                }
                closureColor 
                size{
                  widthInCm
                  weightInKg
                  heightInCm
                  volumeInLiters
                  depthInCm
                }
                additions {
                  lang
                  value
                }
                actualPrice {
                  currency
                  value
                }
                name {
                  lang
                  value
                }
                pattern {
                  lang
                  value
                }
                category {
                  lang
                  value
                }
                quantity
                colors {
                  lang
                  value
                }
                subcategory {
                  lang
                  value
                }
                bottomMaterial {
                  lang
                  value
                }
              }
              delivery {
                sentBy
                invoiceNumber
              }
              paymentMethod
            }
            ...on Error {
              statusCode
              message
            }
          }
        }`,
      variables: {
        id: orderId,
      },
    });

    const order = res.data.getOrderById

    expect(order).toMatchSnapshot()
    expect(order).toMatchObject({
      status:"pending",
      user:{
        __typename: "OrderUser",
        firstName:"Test",
        lastName:"Test",
        email: "test@gmail.com",
        phoneNumber:"380953544271",
        address: {
          __typename: "Address",
          country:"Україна",
          region:"Кіровоградська область",
          city:"Новомиргород",
          zipcode:98908,
          street:"Бульвар Марії Приймаченко",
          buildingNumber:"25",
          appartment:"97"
        }
      },
      delivery:{
        __typename: "Delivery",
        sentBy:"Nova Poshta",
        invoiceNumber:"6280260"
      },
      items:[
        {
          __typename: "OrderItems",
          category:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Сумки"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Bags"
            }
          ],
          subcategory:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Сумки"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Bags"
            }
          ],
          model:
          [
            {
              __typename: "Language",
              lang:"uk",
              value:"Сумка з гобеленом"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Bag with a Pattern"
            }
          ],
          name:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Сумка з гобеленом синя"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Bag with a Pattern Blue"
            }
          ],
          colors:[
            [
              {
                __typename: "Language",
                lang:"uk",
                value:"Сталево-блакитний"
              },
              {
                __typename: "Language",
                lang:"en",
                value:"Steel-blue"
              }
            ]
          ],
          pattern:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Олені"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Deers"
            }
          ],
          closure:[],
          closureColor:"",
          size:{
            __typename: "Size",
            heightInCm:38,
            widthInCm:36,
            depthInCm:10,
            volumeInLiters:0,
            weightInKg:0
          },
          bottomMaterial:[
            {
              __typename: "Language",
              lang:"uk",
              value:"Тканина Кордура"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"Cordura fabric"
            }
          ],
          bottomColor:[
            {
              __typename: "Language",
              lang:"uk",
              value:"чорний"
            },
            {
              __typename: "Language",
              lang:"en",
              value:"black"
            }
          ],
          additions:[],
          actualPrice:[
            {
              __typename: "CurrencySet",
              currency:"UAH",
              value:90000
            },
            {
              __typename: "CurrencySet",
              currency:"USD",
              value:3246
            }
          ],
          quantity:1
        }],
        paymentMethod:"card",
        totalPrice:[
          {
            __typename: "CurrencySet",
            currency:"UAH",
            value:90000
          },
          {
            __typename: "CurrencySet",
            currency:"USD",
            value:3246
          }
        ]
      });
  });

  test('Should throw error ORDER_NOT_FOUND', async () => {
    const res = await client.query({
      query: gql`
        query($id: ID!) {
          getOrderById(id: $id) {
            ...on Order {
              status
              totalPrice {
                currency
                value
              }
              user {
                address {
                  appartment
                  city
                  region
                  street
                  country
                  buildingNumber
                  zipcode
                }
                email
                lastName
                firstName
                phoneNumber
              }
              items {
                bottomColor {
                  lang
                  value
                }
                closure {
                  lang
                  value
                }
                model {
                  lang
                  value
                }
                closureColor 
                size{
                  widthInCm
                  weightInKg
                  heightInCm
                  volumeInLiters
                  depthInCm
                }
                additions {
                  lang
                  value
                }
                actualPrice {
                  currency
                  value
                }
                name {
                  lang
                  value
                }
                pattern {
                  lang
                  value
                }
                category {
                  lang
                  value
                }
                quantity
                colors {
                  lang
                  value
                }
                subcategory {
                  lang
                  value
                }
                bottomMaterial {
                  lang
                  value
                }
              }
              delivery {
                sentBy
                invoiceNumber
              }
              paymentMethod
            }
            ...on Error {
              statusCode
              message
            }
          }
        }`,
      variables: {
        id: "5f46a8ac90e86913ed0a95d8",
      },
    }).catch(err => err)

    const error = res;

    expect(error.graphQLErrors[0].message).toBe("ORDER_NOT_FOUND");
  });

  afterAll(async () => {
    await client.mutate({
      mutation: gql`
        mutation($id: ID!) {
          deleteOrder(id: $id) {
            ... on Order {
              _id
            }
            ... on Error {
              statusCode
              message
            }
          }
        }
      `,
      variables: { id: orderId },
    });
  });
});
