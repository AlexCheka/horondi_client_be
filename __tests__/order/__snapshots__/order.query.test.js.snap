// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order queries should recive order by id 1`] = `
Object {
  "__typename": "Order",
  "delivery": Object {
    "__typename": "Delivery",
    "invoiceNumber": "6280260",
    "sentBy": "Nova Poshta",
  },
  "items": Array [
    Object {
      "__typename": "OrderItems",
      "actualPrice": Array [
        Object {
          "__typename": "CurrencySet",
          "currency": "UAH",
          "value": 90000,
        },
        Object {
          "__typename": "CurrencySet",
          "currency": "USD",
          "value": 3246,
        },
      ],
      "additions": Array [],
      "bottomColor": Array [
        Object {
          "__typename": "Language",
          "lang": "uk",
          "value": "чорний",
        },
        Object {
          "__typename": "Language",
          "lang": "en",
          "value": "black",
        },
      ],
      "bottomMaterial": Array [
        Object {
          "__typename": "Language",
          "lang": "uk",
          "value": "Тканина Кордура",
        },
        Object {
          "__typename": "Language",
          "lang": "en",
          "value": "Cordura fabric",
        },
      ],
      "category": Array [
        Object {
          "__typename": "Language",
          "lang": "uk",
          "value": "Сумки",
        },
        Object {
          "__typename": "Language",
          "lang": "en",
          "value": "Bags",
        },
      ],
      "closure": Array [],
      "closureColor": "",
      "colors": Array [
        Array [
          Object {
            "__typename": "Language",
            "lang": "uk",
            "value": "Сталево-блакитний",
          },
          Object {
            "__typename": "Language",
            "lang": "en",
            "value": "Steel-blue",
          },
        ],
      ],
      "model": Array [
        Object {
          "__typename": "Language",
          "lang": "uk",
          "value": "Сумка з гобеленом",
        },
        Object {
          "__typename": "Language",
          "lang": "en",
          "value": "Bag with a Pattern",
        },
      ],
      "name": Array [
        Object {
          "__typename": "Language",
          "lang": "uk",
          "value": "Сумка з гобеленом синя",
        },
        Object {
          "__typename": "Language",
          "lang": "en",
          "value": "Bag with a Pattern Blue",
        },
      ],
      "pattern": Array [
        Object {
          "__typename": "Language",
          "lang": "uk",
          "value": "Олені",
        },
        Object {
          "__typename": "Language",
          "lang": "en",
          "value": "Deers",
        },
      ],
      "quantity": 1,
      "size": Object {
        "__typename": "Size",
        "depthInCm": 10,
        "heightInCm": 38,
        "volumeInLiters": 0,
        "weightInKg": 0,
        "widthInCm": 36,
      },
      "subcategory": Array [
        Object {
          "__typename": "Language",
          "lang": "uk",
          "value": "Сумки",
        },
        Object {
          "__typename": "Language",
          "lang": "en",
          "value": "Bags",
        },
      ],
    },
  ],
  "paymentMethod": "card",
  "status": "pending",
  "totalPrice": Array [
    Object {
      "__typename": "CurrencySet",
      "currency": "UAH",
      "value": 90000,
    },
    Object {
      "__typename": "CurrencySet",
      "currency": "USD",
      "value": 3246,
    },
  ],
  "user": Object {
    "__typename": "OrderUser",
    "address": Object {
      "__typename": "Address",
      "appartment": "97",
      "buildingNumber": "25",
      "city": "Новомиргород",
      "country": "Україна",
      "region": "Кіровоградська область",
      "street": "Бульвар Марії Приймаченко",
      "zipcode": 98908,
    },
    "email": "test@gmail.com",
    "firstName": "Test",
    "lastName": "Test",
    "phoneNumber": "380953544271",
  },
}
`;
